docs/formats/zdm.md

```diff
@@ -0,0 +1,555 @@
+
+A relational table might look like this:
+```
+{ city: "Berkeley", state: "CA", population: 121643 (uint32) } (=city_schema)
+{ city: "Broad Cove", state: "ME", population: 806 (uint32) } (=city_schema)
+{ city: "Baton Rouge", state: "LA", population: 221599 (uint32) } (=city_schema)
+```
+This ZSON text here depicts three record values.  It defines a type called `city_schema`
+and the inferred type of the `city_schema` has the signature:
+```
+{ city:string, state:string, population:uint32 }
+```
+When all the values in a sequence have the same record type, the sequence
+can be interpreted as a _table_, where the ZSON record values form the _rows_
+and the fields of the records form the _columns_.  In this way, these
+three records form a relational table conforming to the schema `city_schema`.
+
+In contrast, a ZSON text representing a semi-structured sequence of log lines
+might look like this:
+```
+{
+    info: "Connection Example",
+    src: { addr: 10.1.1.2, port: 80 (uint16) } (=socket),
+    dst: { addr: 10.0.1.2, port: 20130 (uint16) } (=socket)
+} (=conn)
+{
+    info: "Connection Example 2",
+    src: { addr: 10.1.1.8, port: 80 (uint16) } (=socket),
+    dst: { addr: 10.1.2.88, port: 19801 (uint16) } (=socket)
+} (=conn)
+{
+    info: "Access List Example",
+    nets: [ 10.1.1.0/24, 10.1.2.0/24 ]
+} (=access_list)
+{ metric: "A", ts: 2020-11-24T08:44:09.586441-08:00, value: 120 }
+{ metric: "B", ts: 2020-11-24T08:44:20.726057-08:00, value: 0.86 }
+{ metric: "A", ts: 2020-11-24T08:44:32.201458-08:00, value: 126 }
+{ metric: "C", ts: 2020-11-24T08:44:43.547506-08:00, value: { x:10, y:101 } }
+```
+In this case, the first records defines not just the a record type
+called `conn`, but also a second embedded record type called `socket`.
+The parenthesized decorators are used where a type is not gleaned from
+the value itself:
+* `socket` is a record with typed fields `addr` and `port` where `port` is an unsigned 16-bit integer, and
+* `conn` is a record with typed fields `info`, `src`, and `dst`.
+
+The subsequent value defines a type called `access_list`.  In this case,
+the `nets` field is an array of networks and illustrates the helpful range of
+primitive types in ZSON.  Note that the syntax here implies
+the type of the array, as it is inferred from the type of the elements.
+
+Finally, there are four more values that show ZSON's efficacy for
+representing metrics.  Here, there are no type decorators as all of the field
+types are implied by their syntax, and hence, the top-level record type is implied.
+For instance, the `ts` field is an RFC 3339 date/time string,
+unambiguously the primitive type `time`.  Further,
+note that the `value` field takes on different types and even a complex record
+type on the last line.  In this case, there is a different type top-level
+record type implied by each of the three variations of type of the `value` field.
+
+## 5. The Zed Data Model
```

Comment:
I kind of think the top 339 lines should be a blog post and from here on should be the contents of the document.

Reply:
A key foundation of Zed is the notion of a "superstructured" type system.  This is foundational to the data model and this text justifies this new idea.  I'm not sure how to just say the Zed data model is a new, superstructured approach to data without explaining it here.

Reply:
Explaining all this is why I created the new document instead of leaving it inside the ZSON spec.